app/helpers/skin_cache_helper.rb

```diff
@@ -1,42 +1,28 @@
 module SkinCacheHelper
-  def skin_cache_html_key(skin, roles_to_include)
-    'skins_style_tags/v3' + \
-      AdminSetting.default_skin.id.to_s + '/' + \
-      Skin.default.id.to_s + '/' + \
-      roles_to_include.to_s + '/' + \
-      skin_cache_value(skin) + '/' + \
-      skin.id.to_s
+  def cache_timestamp
+    Time.now.utc.to_s(:usec)
   end
 
-  def skin_cache_footer_key
-    'Skins_menu/v1/' + \
-    (Rails.cache.fetch('skins_generation/site_skin') || 0).to_s
+  def skin_cache_version_key(skin_id)
+    skin_id = skin_id.id if skin_id.is_a?(Skin)
+    [:v1, :site_skin, skin_id, :version_key]
   end
 
-  def skin_cache_value(skin)
-    (Rails.cache.fetch(skin_memcache_cache_key(skin)) || 0).to_s
+  def skin_cache_version(skin_id)
+    Rails.cache.fetch(skin_cache_version_key(skin_id)) do
+      cache_timestamp
+    end
   end
 
-  def skin_memcache_cache_key(skin)
-    'skins_generation/' + (skin.type.nil? ? ("site_skin_#{skin.id.to_s}") : "work_skin_#{skin.id.to_s}")
+  def skin_cache_version_update(skin_id)
+    Rails.cache.write(skin_cache_version_key(skin_id), cache_timestamp)
   end
 
-  def skin_cache(skin)
-    Rails.cache.increment(skin_memcache_cache_key(skin))
-    Rails.cache.increment('skins_generation/site_skin')  # This is the general key used for the footer
+  def skin_chooser_key
```

Comment:
For this, should we also test that the list of skins in the footer gets updated when a skin is added to/removed from that list?

Reply:
Huh, I thought there was a test for that already, but I guess not! I added some extra steps to the admin test for adding/removing from the chooser.

Reply:
Oh, thanks! I was thinking about manual testing when I asked, but automated tests are good too!

Reply:
Oh, right, manual tests! I added two lines to the Testing Instructions section.

(I'll copy the instructions over to JIRA once this PR is further along. Some of the deploy/testing instructions will depend on how `Skin.get_current_site_skin` is handled, so I don't want to update the issue until that's decided.)